scheduler/roles/slurm/tasks/db.yml

```diff
@@ -0,0 +1,129 @@
+# Copyright 2025 Dell Inc. or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+- name: Is slurm_db_username defined
+  ansible.builtin.set_fact:
+    slurm_db_username: "{{ slurm_db_username_default }}"
+    slurm_db_password: "{{ slurm_db_password_default }}"
+  when: (slurm_db_username is not defined) or (not slurm_db_username)
+
+- name: Is slurm_db_password defined
+  ansible.builtin.set_fact:
+    slurm_db_password: "{{ slurm_db_password_default }}"
+  when: (slurm_db_password is not defined)
+
+- name: Install db packages
+  ansible.builtin.package:
+    name: "{{ db_packages[ansible_os_family][slurm_db_type] }}"
+    state: present 
+  register: new_install
+
+- name: Create my.cnf.d file
+  ansible.builtin.lineinfile:
+    path: /etc/my.cnf
+    line: "[client-server]"
+    create: yes
+
+- name: Include my.cnf.d directory
+  ansible.builtin.lineinfile:
+    path: /etc/my.cnf
+    insertafter: '^\[client-server\]'
+    line: '!includedir /etc/my.cnf.d'
+
+- name: Set db port
+  ansible.builtin.lineinfile:
+    path: "{{ slurm_mysql_cnf_path }}"
+    regexp: '^port=.*'
+    insertafter: '^[mysqld]'
+    line: "port={{ slurm_db_port }}"
+  when: slurm_db_type == 'mysql'
+  notify:
+    - Restart mysqld
```

Comment:
@Cypher-Miller - notifying multiple differnt hosts might not be effective - 
For example: here only restart mysqld and slurmdbd may be triggered, slurmctld may not. Just see if(slurmctld) it useful here. check the other taskd for similar